src/Symfony/Bridge/Doctrine/Tests/DataCollector/DoctrineDataCollectorTest.php

```diff
@@ -25,68 +29,105 @@
 // Doctrine DBAL 2 compatibility
 class_exists(\Doctrine\DBAL\Platforms\MySqlPlatform::class);
 
+/**
+ * @group legacy
```

Comment:
This is the wrong approach to me. Tests using the new API must not be marked as being legacy (meaning you should not use a data provider with a boolean to switch between the old and new APIs)

Reply:
Used a new class for the new api and a trait for common tests